nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ConsumeEWS.java

```diff
@@ -403,7 +420,12 @@ public MimeMessage parseMessage(EmailMessage item) throws Exception {
         //sent date
         mm.setSentDate(ewsMessage.getDateTimeSent());
         //add message headers
-        ewsMessage.getInternetMessageHeaders().forEach(x-> mm.addHeader(x.getName(), x.getValue()));
+        if(hdrList.size() > 0 && hdrList.get(0).equals("*"))
+            ewsMessage.getInternetMessageHeaders().forEach(x-> mm.addHeader(x.getName(), x.getValue()));
+        else if(hdrList.size() > 0)
+            ewsMessage.getInternetMessageHeaders().forEach(x-> {
+                if(hdrList.contains(x.getName())) mm.addHeader(x.getName(), x.getValue());
+            });
```

Comment:
We maybe able to simplify this with:

```
ewsMessage.getInternetMessageHeaders().stream()
    .filter(x -> (hdrList == null || hdrList.isEmpty() || hdrList.contains(x.getName)))
    .forEach(x-> mm.addHeader(x.getName(), x.getValue());
```

Then we could add exclude list as well easily:
```
ewsMessage.getInternetMessageHeaders().stream()
    .filter(x -> (includes == null || includes.isEmpty() || includes.contains(x.getName))
                      && (excludes == null || excludes.isEmpty() || !excludes.contains(x.getName))
    .forEach(x-> mm.addHeader(x.getName(), x.getValue());
```